src/algorithms/l0_smooth.jl

```diff
@@ -0,0 +1,51 @@
+struct L0Smooth <: AbstractImageSmoothAlgorithm
+    """smoothing weight"""
+    λ::Float64
+    """iteration rate"""
+    κ::Float64
+
+    function L0Smooth(λ, κ)
+        λ > zero(λ) || throw(ArgumentError("smoothing weight λ should be positive"))
+        κ > one(κ) || throw(ArgumentError("iteration rate κ should be > 1.0"))
+        new(λ, κ)
+    end
+end
+
+L0Smooth(; λ::Float64=2e-2, κ::Float64=2.0) = L0Smooth(λ, κ)
+
+function (f::L0Smooth)(out::GenericGrayImage,
+                       img::GenericGrayImage)
+    S = float.(channelview(img))

```

Comment:
`of_eltype` from MappedArrays provides a lazy eltype conversion without actually allocating memory.

```suggestion
    S = of_eltype(floattype(eltype(img)), img)
```